File: frontend/src/service/report_service.js
import axios from '../util/axios';
const API_URL = window.location.protocol + '//' + window.location.host + '/api/report/';

class ReportService {
    /**
     * 월 보고서 조회
     * @param {*} ReportRequestDto
     * @returns 
     */
    selectMonthReport(ReportRequestDto) {
        return axios
            .post(API_URL + 'month', ReportRequestDto)
            .then(response => {
                return response.data;
            });
    }
    /**
     * 연간보고서 조회
     * @param {*} ReportRequestDto
     * @returns 
     */
    selectYearReport(ReportRequestDto) {
        return axios
            .post(API_URL + 'year', ReportRequestDto)
            .then(response => {
                return response.data;
            });
    }

}

export default new ReportService();

File: src/main/java/com/app/service/ReportService.java
package com.app.service;

import com.app.dto.LargeCategoryDto;
import com.app.dto.MonthReportDto;
import com.app.dto.ReportRequestDto;
import com.app.dto.YearReportDto;

import java.util.List;

public interface ReportService {
    /**
     * 월 보고서 조회
     * @param reportRequestDto
     * @return MonthReportDto
     */
    MonthReportDto selectMonthReport(ReportRequestDto reportRequestDto);
    /**
     * 연간 보고서 조회
     * @param reportRequestDto
     * @return List<YearReportDto>
     */
    List<YearReportDto> selectYearReport(ReportRequestDto reportRequestDto);
}
